server/routes/events.js

```diff
@@ -12,101 +13,507 @@ router.get('/ping', function(req, res, next) {
+		  							.andWhere((builder) =>
+		  								builder.whereIn('category_id', remove_category_ids)
+		  							)
+										.del();
+		  					}
+		  				})
+							.then(function() {
+								return knex('event_host')
+									.transacting(trx)
+						  		.insert(host_values)
+									.then(function() {
+										return knex('event_host')
+											.transacting(trx)
+											.modify(function(queryBuilder) {
+												if (remove_host_ids){
+													queryBuilder.where({ event_id: event_id })
+														.andWhere((builder) =>
+															builder.whereIn('host_id', remove_host_ids)
+														)
+														.del();
+												}
+											})
+											.then(function() {
+												return knex('event_company')
+													.transacting(trx)
+													.insert(company_values)
+													.then(function() {
+														return knex('event_company')
+															.transacting(trx)
+															.modify(function(queryBuilder){
+																if (remove_company_ids){
+																	queryBuilder.where({ event_id: event_id })
+																		.andWhere((builder) =>
+																			builder.whereIn('company_id', remove_company_ids)
+																		)
+																		.del();
+																}
+															})
+															.then(trx.commit)
+													})
+											})
+									})
+							})
+					})
+  		})
+  		.catch(trx.rollback);
+  })
+  .then(function() {
+  	res.send(util.message('Successfully updated event_id: ' + event_id));
+  })
+  .catch(err => { return next(err) });
+});
+
+
+// Delete an event by event_id
+router.delete('/:event_id', function(req, res, next) {
+  const event_id = req.params.event_id;
+  knex('event').del().where({ fb_id: event_id, }).then(result => {
+    if (result) {
+      res.send(util.message('Successfully deleted event with event_id ' + event_id));
+    } else {
+      util.throwError(404, 'No event found to delete');
+    }
+  })
+  .catch(err => { return next(err) });
 });
 
 // GET all users that have favorited the given event
 router.get('/:event_id/favorites', function(req, res, next) {
   const event_id = req.params.event_id;
-  db.query('SELECT user_id FROM favorite_events WHERE event_id = $1', [event_id], (err, result) => {
-		if (err) return next(err);
-  	res.send(result.rows);
-  });
+  knex('favorite_events').select('user_id')
+  .where({ event_id: event_id })
+  .then(result => {
+    if(result.length) {
+      res.json(result);
+    } else {
+      res.status(404).json('No events where fb_id = ' + event_id + ' found.');
+    }
+  })
 });
 
-// GET all users registered and have paid or not paid for a given event
+// Update user registration for a given event
+router.put('/:event_id/register/:user_id', function(req, res, next) {
+  const event_id = req.params.event_id;
+  const user_id = req.params.user_id;
+  const paid = req.query.paid;
+  knex('event_registration')
+  .update({ 
+    has_paid: paid, 
+    updated_at: knex.raw('now()')
```

Comment:
the updated_at timestamp will automatically update when a row is updated, no need to set it manually